crates/services/txpool_v2/src/service.rs

```diff
@@ -246,36 +169,81 @@ where
 }
 
 #[async_trait::async_trait]
-impl<
-        PSProvider,
-        PSView,
-        ConsensusParamsProvider,
-        GasPriceProvider,
-        WasmChecker,
-        MemoryPool,
-    > RunnableTask
-    for Task<
-        PSProvider,
-        ConsensusParamsProvider,
-        GasPriceProvider,
-        WasmChecker,
-        MemoryPool,
-    >
+impl<View> RunnableTask for Task<View>
 where
-    PSProvider: AtomicView<LatestView = PSView>,
-    PSView: TxPoolPersistentStorage,
-    ConsensusParamsProvider: ConsensusParametersProvider + Send + Sync,
-    GasPriceProvider: GasPriceProviderTrait + Send + Sync,
-    WasmChecker: WasmCheckerTrait + Send + Sync,
-    MemoryPool: MemoryPoolTrait + Send + Sync,
+    View: TxPoolPersistentStorage,
 {
     async fn run(&mut self, watcher: &mut StateWatcher) -> anyhow::Result<bool> {
         let should_continue;
         tokio::select! {
+            biased;
+
             _ = watcher.while_started() => {
                 should_continue = false;
             }
+
+            block_result = self.subscriptions.imported_blocks.next() => {
+                if let Some(result) = block_result {
+                    self.import_block(result);
+                    should_continue = true;
+                } else {
+                    should_continue = false;
```

Comment:
I know the pattern was established before this PR, but it would make more sense to me to just return here instead of declaring and assigning the variable to later return it.

```suggestion
                    return Ok(true);
                } else {
                    return Ok(false);
```

(and the same for all other branches)

Reply:
Nevermind, I see you already merged! 🎉

Reply:
It's Ok you can leave comment and I will address in patch in the future. I note this and let this open in the meantime